docs/root/configuration/listeners/network_filters/generic_proxy_filter.rst

```diff
@@ -0,0 +1,128 @@
+.. _config_network_filters_generic_proxy:
+
+Generic proxy
+=============
+
+
+The network filter could be used to add multiple protocols support to Envoy. The community has implemented lots of this kind of
+network filters, such as :ref:`Dubbo proxy <config_network_filters_dubbo_proxy>`, :ref:`Thrift proxy <config_network_filters_thrift_proxy>`, etc.
+And the developers could also implement their own protocol proxying by creating a new network filter.
+
+
+However, adding new network filter to support new protocol is not easy. The developers need to implement the codec to parse the binary, to implement the
+stream management to handle request/response, to implement the connection management to handle connection lifecycle, etc.
+And there are lots of common features like route, L7 filter chain, tracing, metrics, logging, etc. that need to be implemented again and again in each
+new network filter.
+
+
+Another fact is that lots of RPC protocols has a similar architecture, which is request/response based. The request could be routed to different upstream
+based on the request properties. For these similar protocols, their network filters also are similar and could be abstracted to following modules:
+
+* Codec: parse the binary to request/response object.
+* Stream management: handle the request/response stream.
+* Route: route the request to different upstream based on the request properties.
+* L7 filter chain: filter the request/response.
+* Observability: tracing, metrics, logging, etc.
+
+
+Except the codec, all other modules are common and could be shared by different protocols. This is the motivation of generic proxy.
+
+
+Generic proxy is a network filter that could be used to implement new protocol proxying. An extension point is provided to let the users
+to configure specific codec. The developers only need to implement the codec to parse the binary and let users to configure the codec in the
+generic proxy. The generic proxy will handle the rest of work.
+
+
+Abstract request/response
+-------------------------
+
+
+Abstraction of the request/response is the core of generic proxy. Different L7 protocols may have different request/response data structure.
+But in the generic proxy, except the codec is extented and could be configured by users, other modules are common and shared by different
+protocols. So these different request/response data structures of different protocols need to be abstracted and managed in a common abstraction.
```

Comment:
```suggestion
In the generic proxy, the codec is extended and could be configured by users, but other modules are common and can be shared by different
protocols. These different request/response data structures of different protocols need to be abstracted and managed in a common abstraction.
```

i wasnt totally clear on the meaning here so edited based on my ~understanding

Reply:
I think your understanding is completely right. 😸